pipelines/whale/extractor/metric_runner.py

```diff
@@ -92,42 +92,49 @@ def _get_extract_iter(self):
                 for metric_name, metric_details in metric_yaml.items():
                     execution_time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
 
-                    try:
-                        query = metric_details["sql"]
-                        query = template_query(query, connection_name=database)
-                        result = list(self.execute(query, is_dict_return_enabled=False))
-
-                        try:
-                            cleaned_result = result[0][0]
-                        except:
-                            cleaned_result = None
-
-                        if "description" in metric_details:
-                            description = metric_details["description"]
-                        else:
-                            description = None
-
-                        if "is_global" in metric_details:
-                            is_global = metric_details["is_global"]
-                        else:
-                            is_global = False
-
-                        yield MetricValue(
-                            database=database,
-                            cluster=cluster,
-                            schema=schema,
-                            table=table,
-                            name=metric_name,
-                            description=description,
-                            execution_time=execution_time,
-                            value=cleaned_result,
-                            is_global=is_global,
-                        )
-
-                    except Exception as e:
-                        LOGGER.warning(
-                            f"Failed execution of metric: {metric_name} in {table_stub_path}.\nError: {e}.\nContinuing."
-                        )
+                    sql_result = self._compute_sql_result(metric_details, database)
```

Comment:
Ah so until we get a better solution, we'll need to keep the try/except -- without it, the metrics runner will fail and halt metrics calculations if the SQL is not correctly defined.

Reply:
I am trying to catch that here actually:
https://github.com/dataframehq/whale/blob/1424fa6ef8e251d36adca016410997042d4648e1/pipelines/whale/extractor/metric_runner.py#L128-L134

But still working on that function.

Reply:
Ah got it. Yeah if you can find a good way around it, go for it. :)